aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,104 @@
+The `ResponseCache` attribute specifies the parameters necessary for setting appropriate headers for response caching. The only parameter of the `ResponseCache` attribute that strictly requires the middleware is `VaryByQueryKeys`, which does not correspond to an actual HTTP header. See the [ResponseCache Attribute document](response.md#responsecache-attribute) and the [ResponseCacheAttribute Class API Reference](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute) for usage and a description of the parameters.
+
+## Caching conditions
+* The request must result in a 200 (OK) response from the server.
+* The request method must be GET or HEAD.
+* Terminal middlewares, such as Static File Middleware, must not process the response prior to the Response Caching Middleware.
+* The Authorization header must not be present.
+* `Cache-Control` header parameters must be valid and the response must be marked `public`.
+* The `Pragma: no-cache` header/value must not be present.
+* The `Set-Cookie` header must not be present.
+* `Vary` header parameters must be valid and not equal to `*`.
+
+>[!NOTE]
+> The Antiforgery system for generating secure tokens to prevent Cross-Site Request Forgery (CSRF) attacks will set the `Cache-Control` and `Pragma` headers to `no-cache` so that responses will not be cached.
+
+## HTTP response caching headers
+Response caching by the middleware is configured via your HTTP response headers. The relevant headers are listed below with notes on how they affect caching.
+
+Header | Details
+--- | --- |
+Authorization | <p>The response is not cached if the header exists.</p>
+Cache-Control | <p>The header must be present and its value must explicitly be marked `public`.</p><p>You can control caching with the following parameters:</p><ul><li>max-age</li><li>max-stale</li><li>min-fresh</li><li>must-revalidate</li><li>no-cache</li><li>no-store</li><li>only-if-cached</li><li>private</li><li>public</li><li>s-maxage</li><li>proxy-revalidate</li></ul>
+Pragma | <p>The value must not be `no-cache`.</p><p>Considered for backward compatibility with HTTP/1.0.</p>
+Set-Cookie | <p>The response is not cached if the header exists.</p>
+Vary | <p>You can vary the cached response by another header. For example, you can cache responses by encoding by including the `Vary: Accept-Encoding` header, which would cache `gzip` and `text/plain` responses separately.</p>
+Expires | <p>The response must not be stale given the value provided.</p>
+If-None-Match | <p>The response will be served from cache if the value is not `*` and the `ETag` of the response doesn't match any of the values provided. Otherwise, a 304 (Not Modified) response will be served.</p>
+If-Modified-Since | <p>If the `If-None-Match` header is not present, a cached response will be served if the cached response date value is less than the value provided.</p>
+
+The middleware sets the following headers when appropriate:
```

Comment:
I suppose it's better to say that `Date` and `Content-Length` will be set by the middleware when serving from cache if they are not provided on the original response. `Age` sent in the original response will be ignored and the middleware will compute a new value when serving.